modules/dnn/src/layers/recurrent_layers.cpp

```diff
@@ -90,16 +90,49 @@ class LSTMLayerImpl : public LSTMLayer
 
     bool useTimestampDim;
     bool produceCellOutput;
+    float forgetBias, cellClip;
+    bool useCellClip, usePeephole;
 
 public:
 
     LSTMLayerImpl(const LayerParams& params)
         : numTimeStamps(0), numSamples(0)
     {
         setParamsFrom(params);
-        type = "LSTM";
-        useTimestampDim = true;
-        produceCellOutput = false;
+
+        if (!blobs.empty())
+        {
+            CV_Assert(blobs.size() >= 3);
+
+            blobs[2] = blobs[2].reshape(1, 1);
+
+            const Mat& Wh = blobs[0];
+            const Mat& Wx = blobs[1];
+            const Mat& bias = blobs[2];
+            CV_Assert(Wh.dims == 2 && Wx.dims == 2);
+            CV_Assert(Wh.rows == Wx.rows);
+            CV_Assert(Wh.rows == 4*Wh.cols);
+            CV_Assert(Wh.rows == (int)bias.total());
+            CV_Assert(Wh.type() == Wx.type() && Wx.type() == bias.type());
+
+            // Peephole weights.
+            if (blobs.size() > 3)
+            {
+                CV_Assert(blobs.size() == 6);
+                for (int i = 3; i < 6; ++i)
+                {
+                    CV_Assert(blobs[i].rows == Wh.cols && blobs[i].cols == Wh.cols);
```

Comment:
@dkurt , shouldn't it be `blobs[i].rows == Wh.rows && ...`?

Reply:
@mshabunin, looks like these blobs are squad. Actually, they are diagonal matrices: https://github.com/opencv/opencv/pull/9491/files#diff-c6e561028becbce9ec212c49506ce60dR1150.  Is there any memory problems?

Reply:
Just a warning from the static analyzer:
```
>>>     CID 1420444:  Incorrect expression  (COPY_PASTE_ERROR)
>>>     "cols" in "this->blobs[i].rows == Wh.cols" looks like a copy-paste error.
124                         CV_Assert(blobs[i].rows == Wh.cols && blobs[i].cols == Wh.cols);
```
Probably it can be suppressed by rewriting the condition: `CV_Assert(blobs[i].rows == blobs[i].cols && blobs[i].cols == Wh.cols);`
I will look into it, thank you.